addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2644,33 +2720,26 @@ export class OdooEditor extends EventTarget {
         if (anchorNode && !ancestors(anchorNode).includes(this.editable)) {
             return false;
         }
+        const selectSingleCell  = this._currentKeyPress?.isShiftKey && this._wasTableSelected;
+        delete this._wasTableSelected;
         const traversedNodes = getTraversedNodes(this.editable);
         if (this._isResizingTable || !traversedNodes.some(node => !!closestElement(node, 'td') && !isProtected(node))) {
             return false;
         }
-        let range;
-        if (selection.rangeCount > 1) {
-            // Firefox selection in table works with multiple ranges.
-            const startRange = getDeepRange(this.editable, {range: selection.getRangeAt(0)});
-            const endRange = getDeepRange(this.editable, {range: selection.getRangeAt(selection.rangeCount - 1)});
-            range = this.document.createRange();
-            range.setStart(startRange.startContainer, 0);
-            range.setEnd(endRange.startContainer, 0);
-        } else {
-            // We need the triple click correction only for a bug in firefox
-            // where it gives a selection of a full cell as tr 0 tr 1. The
-            // correction makes it so it gives us the cell and not its neighbor.
-            // In all other cases we don't want to make that correction so as to
-            // avoid flicker when hovering borders.
-            range = getDeepRange(this.editable, { correctTripleClick: anchorNode && anchorNode.nodeName === 'TR' });
-        }
+        // Firefox selection in table works with multiple ranges.
+        // We need the triple click correction only for a bug in firefox
+        // where it gives a selection of a full cell as tr 0 tr 1. The
+        // correction makes it so it gives us the cell and not its neighbor.
+        // In all other cases we don't want to make that correction so as to
+        // avoid flicker when hovering borders.
+        const range = handleMultipleRanges(this.editable, { correctTripleClick: selection.anchorNode && selection.anchorNode.nodeName === 'TR' });
```

Comment:
I see you have added `selection.anchorNode && selection.anchorNode.nodeName === 'TR'` at all places where `handleMultipleRanges` is used. Can't we fix this within `handleMultipleRanges` function itself rather than adding it everywhere?